libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -635,6 +635,116 @@ string YulUtilFunctions::overflowCheckedIntExpFunction(
 	});
 }
 
+string YulUtilFunctions::overflowCheckedIntLiteralExpFunction(
+	RationalNumberType const& _baseType,
+	IntegerType const& _exponentType,
+	IntegerType const& _commonType
+)
+{
+	solAssert(!_exponentType.isSigned(), "");
+	solAssert(_baseType.isNegative() == _commonType.isSigned(), "");
+	solAssert(_commonType.numBits() == 256, "");
+
+	// Calculates the upperbound for exponentiation, that is, calculate `b`, such that
+	// _base**b <= _maxValue and _base**(b + 1) > _maxValue
+	auto findExponentUpperbound = [](bigint const _base, bigint const _maxValue) -> unsigned
+	{
+		unsigned first = 0;
+		unsigned last = 255;
+		unsigned middle;
+
+		while (first < last)
+		{
+			middle = (first + last) / 2;
+
+			if (
+				// The condition on msb is a shortcut that avoids computing large powers in
+				// arbitrary precision.
+				boost::multiprecision::msb(_base) * middle <= boost::multiprecision::msb(_maxValue) &&
+				boost::multiprecision::pow(_base, middle) <= _maxValue
+			)
+			{
+				if (boost::multiprecision::pow(_base, middle + 1) > _maxValue)
+					return middle;
+				else
+					first = middle + 1;
+			}
+			else
+				last = middle;
+		}
+
+		return last;
+	};
+
+
+	string functionName = "checked_exp_" + _baseType.richIdentifier() + "_" + _exponentType.identifier();
+	u256 baseValue = _baseType.literalValue(nullptr);
+	unsigned exponentUpperbound;
+
+	// Bases 0, 1 and -1. There is no overflow or underflow. The optimizer can simplify the `exp`
```

Comment:
For -1, the simplifcation is implemented https://github.com/ethereum/solidity/pull/9895

Reply:
What is the use case of `-1**n`? What code would produce that?

Reply:
Line 694. The `_baseValue` is `2**256 - 1` (i.e., -1)

Reply:
If you mean what solidity code would produce it, then I cannot think of a good example.

Reply:
I think realistically only `2**n` and `10**n` occur in contracts, except some fringe edge cases.

Reply:
We already have optimizations for `0`, `1` and `2` (during compile time) and `0, 1, 2, 10` and `256` (during runtime)